project5/project5.cc

```diff
@@ -0,0 +1,442 @@
+                            break;
+                        }
+                    }
+                }
+
+                //As long as the road isn't connecting two towns that have already
+                //been connected, we add it to the minimum spanning tree
+                if(!(component[startTown] == 1 && component[visitTown] == 1))
+                {
+                    size++;
+                    component[startTown] = 1;
+                    component[visitTown] = 1;
+                    output << "\t" << _towns[startTown]._name << " to " << _towns[visitTown]._name << endl;
+                }
+            }
+            if(_numberOfTowns == 1)
+                output << "\t" << _towns[0]._name << " is the only town in the whole country!"<<endl;
+        }
+
+        //looks for any articulation points in the country graph
+        //outputting the name of the town if it is an articulation point
+        //outputting none if there are no articulation points
+        void CountryGraph::findArticulationPoints(ostream & output)
+        {
+
+            cout << "\nObliterating any of the following would result in the province becoming disconnected:\n" << endl;
+
+            //bools for the towns visited and articulationPoints found
+            bool visited[_numberOfTowns];
+            bool articPoint[_numberOfTowns];
+
+            //Set all bools to false
+            for(int i =0; i < _numberOfTowns; i++)
+            {
+                visited[i] = false;
+                articPoint[i] = false;
+            }
+
+            //ints for the low values, num values, and the parent indexes
+            int num[_numberOfTowns];
+            int low[_numberOfTowns];
+            int parent[_numberOfTowns];
+            int counter = 0;
+
+            //Set all parents to -1
+            for(int i =0; i < _numberOfTowns; i++)
+            {
+                parent[i] = -1;
+            }
+
+            //calls the auxilary function for finding articulation points on each
+            //unvisited town
+            for(int i = 0; i < _numberOfTowns; i++)
+            if(! visited[i])
+            findArticulationPointsAUX(i, visited, articPoint, num, low, parent, counter);
+
+            bool foundArticPoints = false;
+
+            // outputs all the articulation points and sets foundArticPoints to true
+            for(int i = 0; i < _numberOfTowns; i++)
+            if(articPoint[i])
+            {
+                output << "\t" << _towns[i]._name << endl;
+                foundArticPoints = true;
+            }
+
+            //if there were no Articulation points, output none
+            if(!foundArticPoints)
+            output << "\tNone" << endl;
+        }
+
+        // the auxilary function for finding articulation points on the unvisited towns
+        void CountryGraph::findArticulationPointsAUX(int currentTown, bool visited[], bool articPoint[], int num[], int low[], int parent[], int counter)
```

Comment:
Good method.  Prolog comment should describe the arguments. Does it need that many arguments?  Some seem like they could be local variables instead.